### PR TITLE
perf: get event types

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

improves performance of get event types

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1) Go to /event-types

### CODE REVIEW

File: packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts [hunk 1]
```diff
@@ -107,7 +98,17 @@ const fetchEventTypesBatch = async (
...
+    ]);
+
+    const userEventTypes = [...(nonChildEventTypes ?? []), ...(childEventTypes ?? [])].sort((a, b) => {
+      // First sort by position in descending order
+      if (a.position !== b.position) {
+        return b.position - a.position;
+      }
+      // Then by id in ascending order
+      return a.id - b.id;
+    });
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts:111]: parallel queries could alter the positons

File: packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts [hunk 2]
```diff
@@ -147,48 +148,11 @@ const fetchEventTypesBatch = async (
...
-      return true;
-    }
-    // A child event only has one user
-    const childEventAssignee = eventType.users[0];
-
-    if (!childEventAssignee || childEventAssignee.id !== userId) {
-      return false;
-    }
-    return true;
-  });
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts:179]: moved this to prisma query

**Review by sean-brydon — APPROVED** [review]
LGTM nice one! Tested and it works, much faster than the other approach from what i can see in dev

File: packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts [hunk 3]
```diff
@@ -37,64 +38,33 @@ export const getEventTypesFromGroup = async ({
...
-    const userEventTypes =
-      (await EventTypeRepository.findAllByUpId(
+    const baseQueryConditions = {
+      teamId: null,
+      schedulingType: null,
+      ...(searchQuery ? { title: { contains: searchQuery, mode: "insensitive" as Prisma.QueryMode } } : {}),
+    };
+
+    const [nonChildEventTypes, childEventTypes] = await Promise.all([
+      EventTypeRepository.findAllByUpId(
```
[the remark this answers is not in the document]
Comment (keithwillcode) [reply packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts:59]: Seems fine to split

**Pull request merged**